docs/source/checks/vision/data_integrity/plot_property_label_correlation.py

```diff
@@ -0,0 +1,137 @@
+abstractions. For example, in the dataset of wolves and dogs photographs, the brightness of the image may be used to
+predict the label "wolf" easily.
+
+
+How do we calculate for different vision tasks?
+-----------------------------------------------
+
+* For classification tasks, this check uses PPS to predict the class by image properties.
+* For object detection tasks, this check uses PPS to predict the class of each
+  bounding box, by the image properties of that specific bounding box.
+  This means that for each image, this check crops all the sub-images defined by bounding
+  boxes, and uses them as inputs as though they were regular classification dataset images.
+
+How is the Predictive Power Score (PPS) calculated?
+---------------------------------------------------
+The properties' predictive score results in a numeric score between 0 (feature has
+no predictive power) and 1 (feature can fully predict the label alone).
+
+The process of calculating the PPS is the following:
+
+The properties' predictive score results in a numeric score between 0 (feature has no predictive power) and 1 (feature
+can fully predict the label alone).
+
+The process of calculating the PPS is the following:
+"""
+#%%
+# 1. Extract from the data only the label and the feature being tested
+# 2. Drop samples with missing values
+# 3. Keep 5000 (this is configurable parameter) samples from the data
+# 4. Preprocess categorical columns. For the label using `sklearn.LabelEncoder` and for the feature
+#    using `sklearn.OneHotEncoder`
+# 5. Partition the data with 4-fold cross-validation
+# 6. Train decision tree
+# 7. Compare the trained model's performance with naive model's performance as follows:
+#
+# Regression: The naive model always predicts the median of the label column,
+# the metric being used is MAE and the PPS calculation is: :math:`1 - \frac{\text{MAE model}}{\text{MAE naive}}`
+#
+# Classification: The naive model always predicts the most common class of
+# the label column, The metric being used is F1 and the PPS calculation is:
+# :math:`\frac{\text{F1 model} - \text{F1 naive}}{1 - \text{F1 naive}}`
+#
+# .. note::
+#
+#    All the PPS parameters can be changed by passing to the check the parameter ``ppscore_params``
+#
+#
+# For further information about PPS you can visit the `ppscore github <https://github.com/8080labs/ppscore>`__
+# or the following blog post: `RIP correlation. Introducing the Predictive Power Score
+# <https://towardsdatascience.com/rip-correlation-introducing-the-predictive-power-score-3d90808b9598>`__
+
+#%%
+# Run the Check
+# =============
+# In this example we will run the check on the dataset of wolves vs. dogs.
+# For example purposes we picked 10 images of dogs and 10 images of wolves out of the full dataset.
+# The original data was downloaded from https://www.kaggle.com/datasets/harishvutukuri/dogs-vs-wolves, which is licensed under `DbCL v1.0 <https://opendatacommons.org/licenses/dbcl/1-0/>`__.
```

Comment:
from my perspective we can drop the last part (starting from "which is license"..). Since this is our responsibility to check we use it properly, but it's not likely less interesting for the user what's the license type. (unless the license terms state that it should be included or something).

Reply:
@noamzbr you asked me to add the license part so your call

Reply:
"You must attribute any public use of the contents, or works produced from them. For redistribution or derivative works, you must make the use of this license clear to others, and you must keep intact any notices on the original"
@shir22 No way around this